### PR TITLE
Add web_microphone_websocket example

### PR DESCRIPTION
**Pull request by dsteinman**
This is a JavaScript example which streams a web microphone recording over a websocket to NodeJS which streams it into DeepSpeech with VAD processing. The results from DeepSpeech are sent back to the client.

added unit tests for both client and server:
yarn run test:client
yarn run test:server

I added those to the `test.sh` file but didn't actually try that file out because I don't know how it hooks into DeepSpeech tools.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -16,11 +16,12 @@ It is a good way to just try out DeepSpeech, without much knowledge on how DeepS
 
 * `Microphone VAD streaming  <mic_vad_streaming/README.rst>`_
 * `VAD transcriber  <vad_transcriber/>`_
-
+ss
```
Comment (lissyx) [inline README.rst]: Looks like another small typo :)

**Comment by lissyx** [comment]
Thanks @dsteinman I'll try and run that against CI, but it looks good. I'll have a closer look tomorrow

File: README.rst (hunk 1 above)
Comment (dsteinman) [reply README.rst]: Oops yeah, I've just removed it now.

**Comment by dsteinman** [comment]
I just realized this example does not work in Firefox.  I'll try to get it working.

**Comment by dsteinman** [comment]
Okay, it's fixed in Firefox, there's a 2nd commit now with the changes.

**Comment by reuben** [comment]
This is awesome, thanks for the contribution!

**Pull request merged**